### PR TITLE
Add Native Components guide.

### PR DESCRIPTION
**Pull request by sahrens**

### CODE REVIEW

File: docs/NativeComponentsIOS.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+Like the native module guide, this too is a more advanced guide that assumes you are somewhat familiar with iOS programming.  This guide will show you how to build a native UI component, walking you through the implementation of a subset of the existing `MapView` component available in the core React Native library.
+
+## iOS MapView example
+
+Let's say we want to add an interactive Map to our app - might as well use [`MKMapView`](https://developer.apple.com/library/prerelease/mac/documentation/MapKit/Reference/MKMapView_Class/index.html), we just need to make it usable from JavaScript.
+
+Native views are created and manipulated by subclasses of `RCTViewManager`.  These subclasses are similar in function to view controllers, but are essentially singletons - only one instance of each is created by the bridge.  They vend native views to the `RCTUIManager`, which delegates back to them to set and update the properties of the views as necessary.  The `RCTViewManager`s are also typically the delegates for the views, sending events back to JavaScript via the bridge.
```
Comment (brentvatne) [inline docs/NativeComponentsIOS.md]: "they vend" -> "they send"

Comment (brentvatne) [inline docs/NativeComponentsIOS.md]: scratch that, read the next paragraph and it makes sense ;P

File: docs/NativeComponentsIOS.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+var { requireNativeComponent } = require('react-native');
+
+module.exports = requireNativeComponent('RCTMap', null);
+```
+
+This is now a fully-functioning native map view component in JavaScript, complete with pinch-zoom and other native gesture support.  We can't really control it from JavaScript yet, though :(
+
+## Properties
+
+The first thing we can do to make this component more usable is to bridge over some native properties. Let's saw we want to be able to disable pitch control, and be able to specify the visible region.  Disabling pitch is a simple boolean, so we just add this to the m-file:
```
Comment (brentvatne) [inline docs/NativeComponentsIOS.md]: "Let's saw" -> "Let's say"

**Comment by brentvatne** [comment]
Great job @sahrens - this will help a lot of people!

**Comment by sahrens** [comment]
Yeah, been meaning to write this for a while but wanted requireNativeComponent to land first ;)

**Pull request merged**